shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -1080,6 +1080,19 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool, var ne
                 err(msg"Class ${name} has no method ${fieldName.name}", term.toLoc)
                 mthCallOrSel(obj, fieldName)
             }
+          // * The code below is only a temporary solution to type `ClassName.unapply`.
+          // * It removed when static methods can be typed properly.
+          case Var(nuCls) =>
+            if (fieldName.name === "unapply") (ctx.get(nuCls) match {
+                case S(CompletedTypeInfo(cls: TypedNuCls)) => cls.td.genUnapply
+                case S(ti: DelayedTypeInfo) => ti.decl.genUnapply
+                case _ => N
+              }) match {
+                case S(NuFunDef(_, _, _, L(unapplyMtd))) =>
+                  typePolymorphicTerm(unapplyMtd)
```

Comment:
This would better be
```suggestion
                  typeMonomorphicTerm(unapplyMtd)
```
because the result is applied immediately (no need to have it be generalized/polymorphic).

Reply:
Never mind, I realized that `genUnapply` actually returns a lambda with an unbound type variable occurrence, and the latter must be generalized before the result can be used.